src/audio/module_adapter/module/volume/volume_generic.c

```diff
@@ -173,22 +180,26 @@ static void vol_s32_to_s32(struct comp_dev *dev, struct audio_stream *sink,
  * Copy and scale volume from 16 bit source buffer
  * to 16 bit destination buffer.
  */
-static void vol_s16_to_s16(struct comp_dev *dev, struct audio_stream *sink,
-			   const struct audio_stream *source, uint32_t frames)
+static void vol_s16_to_s16(struct processing_module *mod, struct input_stream_buffer *bsource,
+			   struct output_stream_buffer *bsink, uint32_t frames, uint16_t nch)
 {
-	struct vol_data *cd = comp_get_drvdata(dev);
+	struct vol_data *cd = module_get_private_data(mod);
+	struct audio_stream *source = bsource->data;
+	struct audio_stream *sink = bsink->data;
 	int32_t vol;
 	int16_t *x, *x0;
 	int16_t *y, *y0;
 	int nmax, n, i, j;
-	const int nch = source->channels;
 	int remaining_samples = frames * nch;
 #if CONFIG_COMP_PEAK_VOL
 	int16_t tmp = INT_MIN(16);
 #endif
 
 	x = source->r_ptr;
 	y = sink->w_ptr;
+
+	bsource->consumed += remaining_samples << 1;
+	bsink->size += remaining_samples << 1;
```

Comment:
Ditto for s16 volume.

Could also with additional commit clean up functions names like vol_s16_to_s16() to vol_s16() since the format conversion has been removed long ago. In volume processing functions the input and output format are always the same.

Reply:
sure, I will do this ins a follow up PR